mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.java

```diff
@@ -0,0 +1,282 @@
+                conn.setUsername(getUsername());
+            }
+            if (hasPassword()) {
+                conn.setPassword(getPassword());
+            }
+
+            return conn;
+        } else {
+            return null;
+        }
+    }
+
+    private String getPassword() {
+        return settings.getString(Constants.PREFERENCE_LOCAL_PASSWORD, null);
+    }
+
+    private boolean hasPassword() {
+        return getPassword() != null && !getPassword().isEmpty();
+    }
+
+    private String getUsername() {
+        return settings.getString(Constants.PREFERENCE_LOCAL_USERNAME, null);
+    }
+
+    private boolean hasUsername() {
+        return getUsername() != null && !getUsername().isEmpty();
+    }
+
+    private Connection getRemoteConnection() {
+        String openHABUrl = Util.normalizeUrl(settings.getString(Constants
+                .PREFERENCE_ALTURL, ""));
+        // If remote URL is configured
+        if (openHABUrl.length() > 0) {
+            Log.d(TAG, "Connecting to remote URL " + openHABUrl);
+            Connection conn = new DefaultConnection(ctx);
+            conn.setOpenHABUrl(openHABUrl);
+            conn.setConnectionType(Connection.TYPE_REMOTE);
+
+            if (hasRemoteUsername()) {
+                conn.setUsername(getRemoteUsername());
+            }
+            if (hasRemotePassword()) {
+                conn.setPassword(getRemotePassword());
+            }
+
+            return conn;
+        } else {
+            return null;
+        }
+    }
+
+    private String getRemotePassword() {
+        return settings.getString(Constants.PREFERENCE_REMOTE_PASSWORD, null);
+    }
+
+    private boolean hasRemotePassword() {
+        return getRemotePassword() != null && !getRemotePassword().isEmpty();
+    }
+
+    private String getRemoteUsername() {
+        return settings.getString(Constants.PREFERENCE_REMOTE_USERNAME, null);
+    }
+
+    private boolean hasRemoteUsername() {
+        return getRemoteUsername() != null && !getRemoteUsername().isEmpty();
+    }
+
+    private Connection getAvailableConnection() {
+        ConnectivityManager connectivityManager =
+                (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
+        if (activeNetworkInfo != null) {
+            if (settings.getBoolean(Constants.PREFERENCE_DEMOMODE, false)) {
+                Log.d(TAG, "Demo mode");
+
+                return new DemoConnection(ctx);
+            } else {
+                // If we are on a mobile network go directly to remote URL from settings
+                if (activeNetworkInfo.getType() == ConnectivityManager.TYPE_MOBILE) {
+                    if (getRemoteConnection() == null) {
+                        throw new NoUrlInformationException();
+                    }
+                    return getRemoteConnection();
+
+                    // Else if we are on Wifi or Ethernet network
```

Comment:
Please add the other conn types in the comment

Reply:
Done!